ios/Handlers/ResourcesHandler.swift

```diff
@@ -1,137 +1,82 @@
-                    completion(nil, error)
-                }
-            }
-        }
     }
-
-    fileprivate func implicitResourcesRequest(_ details: NSDictionary, _ completion: @escaping (String?, Error?) -> Void) {
+    
+    private func buildRequest(_ details: NSDictionary) throws -> ResourceRequest {
         guard let path = details["path"] as? String else {
-            let error = NSError(domain: ONGFetchImplicitResourceErrorDomain, code: 0, userInfo: [NSLocalizedDescriptionKey : "'path' can not be empty"])
-            completion(nil, error)
-            return
+            throw WrapperError.parametersNotCorrect(description: "'path' can not be empty")
         }
-        guard let method = details["method"] as? String else {
-            let error = NSError(domain: ONGFetchImplicitResourceErrorDomain, code: 0, userInfo: [NSLocalizedDescriptionKey : "'method' can not be empty"])
-            completion(nil, error)
-            return
+        guard let methodName = details["method"] as? String,
+              let method = HTTPMethod(rawValue: methodName.uppercased())
+        else {
+            throw WrapperError.parametersNotCorrect(
+                description: "'method' must be either 'GET', 'POST', 'PUT', 'DELETE', 'PATCH' or 'HEAD'")
         }
         
-        let implicitRequest = ONGResourceRequest(path: path, method: method, parameters: details["parameters"] as? [String : Any], encoding: ONGParametersEncoding.formURL, headers: details["headers"] as? [String : String]);
-
-        ONGUserClient.sharedInstance().fetchImplicitResource(implicitRequest) { response, error in
-            if let error = error {
-                completion(nil, error)
-            } else {
-                if let data = response?.data {
-                    completion(String(data: data, encoding: .utf8), nil)
-                } else {
-                    let error = NSError(domain: ONGFetchImplicitResourceErrorDomain, code: 0, userInfo: [NSLocalizedDescriptionKey : "Response doesn't contain data."])
-                    completion(nil, error)
-                }
+        let headers = try getHeaders(details)
+        return ResourceRequestFactory.makeResourceRequest(
+            path: path, method: method, parameters: nil, body: nil,
+            headers: headers, parametersEncoding: .formURL)
+    }
+    
+    private func getHeaders(_ details: NSDictionary) throws -> [String: String] {
+        guard let headers = details["headers"] as? [String: Any] else {
+            throw WrapperError.parametersNotCorrect(
+                description: "'headers' must be an object of String: String")
+        }
+        var result: [String: String] = [:]
+        for (key, value) in headers {
+            if let value = value as? String {
+                result[key] = value
             }
         }
+        return result
```

Comment:
So basically you want to have all `String` values out of `headers`?

Reply:
yes